component/src/main/java/com/yahoo/component/VersionCompatibility.java

```diff
@@ -0,0 +1,109 @@
+// Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+package com.yahoo.component;
+
+import java.util.List;
+import java.util.NavigableMap;
+import java.util.TreeMap;
+
+/**
+ * Logic for what platform and compile versions are incompatible.
+ *
+ * This class is instantiated with a list of versions which are only compatible with other versions
+ * that are not older than themselves. Each entry in this list is represented by a string, and any
+ * major, minor or micro with the value @{code *} means all versions are such boundaries.
+ * Versions A, B are incompatible iff. there exists an incompatibility boundary X such that
+ * (A &ge; X) &ne; (B &ge; X).
+ *
+ * @author jonmv
+ */
+public class VersionCompatibility {
```

Comment:
Why do we need this flexibility in verifying incompatible versions? Seeing as we use semantic versioning I don't think we ever need to express incompatibility between 1.2.3 and 1.x.x? Major is sufficient.

Reply:
The intention was that we could set a different range (`List.of("*.*.*")`) for a test app, in our CD tests.

Reply:
If we set that for a particular application, the system will treat all its packages as incompatible with other platform versions than the one used to compile the package, and we'll have a test for this that runs all the time. I don't think there's a cleaner way to do that.

Reply:
I see, good idea 🙂

Reply:
Well, if there's only that special value we need, we could of course simplify this a lot ... >_<